Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def actual_url(url)
+    m = Utils.popen_read("curl", "-I", "-L", url).match /^Location: (.+)$/
```

Comment:
Yes, the test may create false positive as there aren't many resources with multiple redirects.